// @radix-ui/react-form@0.1.2/dist/index.js
import * as o from "react";
import { composeEventHandlers as S } from "../../../../../@radix-ui_primitive@1.1.1/node_modules/@radix-ui/primitive/dist/index.js";
import { useComposedRefs as N } from "../../../../../@radix-ui_react-compose-refs@1.1.1_@types_react@19.0.10_react@19.0.0/node_modules/@radix-ui/react-compose-refs/dist/index.js";
import { createContextScope as de } from "../../../../../@radix-ui_react-context@1.1.1_@types_react@19.0.10_react@19.0.0/node_modules/@radix-ui/react-context/dist/index.js";
import { useId as U } from "../../../../../@radix-ui_react-id@1.1.0_@types_react@19.0.10_react@19.0.0/node_modules/@radix-ui/react-id/dist/index.js";
import { Label as ue } from "../../../../../@radix-ui_react-label@2.1.2_@types_react-dom@19.0.4_@types_react@19.0.10__@types_react@_6292b41e3d1ceb1104e2dc903fb790d1/node_modules/@radix-ui/react-label/dist/index.js";
import { Primitive as j } from "../../../../../@radix-ui_react-primitive@2.0.2_@types_react-dom@19.0.4_@types_react@19.0.10__@types_re_eba4716596d4f139c393e6187e0b8922/node_modules/@radix-ui/react-primitive/dist/index.js";
import { j as h } from "../../../../../../../_virtual/jsx-runtime.js";
var [q, Pe] = de("Form"), G = "Form", [me, L] = q(G), [fe, W] = q(G), X = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, onClearServerErrors: c = () => {
    }, ...a } = e, l = o.useRef(null), f = N(t, l), [d, v] = o.useState({}), u = o.useCallback(
      (r) => d[r],
      [d]
    ), p = o.useCallback(
      (r, m) => v((i) => ({
        ...i,
        [r]: { ...i[r] ?? {}, ...m }
      })),
      []
    ), x = o.useCallback((r) => {
      v((m) => ({ ...m, [r]: void 0 })), n((m) => ({ ...m, [r]: {} }));
    }, []), [C, y] = o.useState({}), I = o.useCallback(
      (r) => C[r] ?? [],
      [C]
    ), M = o.useCallback((r, m) => {
      y((i) => ({
        ...i,
        [r]: [...i[r] ?? [], m]
      }));
    }, []), g = o.useCallback((r, m) => {
      y((i) => ({
        ...i,
        [r]: (i[r] ?? []).filter(
          (E) => E.id !== m
        )
      }));
    }, []), [_, n] = o.useState({}), F = o.useCallback(
      (r) => _[r] ?? {},
      [_]
    ), b = o.useCallback((r, m) => {
      n((i) => ({
        ...i,
        [r]: { ...i[r] ?? {}, ...m }
      }));
    }, []), [R, A] = o.useState({}), w = o.useCallback((r, m) => {
      A((i) => {
        const E = new Set(i[r]).add(m);
        return { ...i, [r]: E };
      });
    }, []), T = o.useCallback((r, m) => {
      A((i) => {
        const E = new Set(i[r]);
        return E.delete(m), { ...i, [r]: E };
      });
    }, []), k = o.useCallback(
      (r) => Array.from(R[r] ?? []).join(" ") || void 0,
      [R]
    );
    return /* @__PURE__ */ h.jsx(
      me,
      {
        scope: s,
        getFieldValidity: u,
        onFieldValidityChange: p,
        getFieldCustomMatcherEntries: I,
        onFieldCustomMatcherEntryAdd: M,
        onFieldCustomMatcherEntryRemove: g,
        getFieldCustomErrors: F,
        onFieldCustomErrorsChange: b,
        onFieldValiditionClear: x,
        children: /* @__PURE__ */ h.jsx(
          fe,
          {
            scope: s,
            onFieldMessageIdAdd: w,
            onFieldMessageIdRemove: T,
            getFieldDescription: k,
            children: /* @__PURE__ */ h.jsx(
              j.form,
              {
                ...a,
                ref: f,
                onInvalid: S(e.onInvalid, (r) => {
                  const m = re(r.currentTarget);
                  m === r.target && m.focus(), r.preventDefault();
                }),
                onSubmit: S(e.onSubmit, c, {
                  checkForDefaultPrevented: !1
                }),
                onReset: S(e.onReset, c)
              }
            )
          }
        )
      }
    );
  }
);
X.displayName = G;
var H = "FormField", [ve, K] = q(H), Y = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, name: c, serverInvalid: a = !1, ...l } = e, d = L(H, s).getFieldValidity(c), v = U();
    return /* @__PURE__ */ h.jsx(ve, { scope: s, id: v, name: c, serverInvalid: a, children: /* @__PURE__ */ h.jsx(
      j.div,
      {
        "data-valid": z(d, a),
        "data-invalid": J(d, a),
        ...l,
        ref: t
      }
    ) });
  }
);
Y.displayName = H;
var O = "FormLabel", Z = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, ...c } = e, a = L(O, s), l = K(O, s), f = c.htmlFor || l.id, d = a.getFieldValidity(l.name);
    return /* @__PURE__ */ h.jsx(
      ue,
      {
        "data-valid": z(d, l.serverInvalid),
        "data-invalid": J(d, l.serverInvalid),
        ...c,
        ref: t,
        htmlFor: f
      }
    );
  }
);
Z.displayName = O;
var D = "FormControl", ee = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, ...c } = e, a = L(D, s), l = K(D, s), f = W(D, s), d = o.useRef(null), v = N(t, d), u = c.name || l.name, p = c.id || l.id, x = a.getFieldCustomMatcherEntries(u), { onFieldValidityChange: C, onFieldCustomErrorsChange: y, onFieldValiditionClear: I } = a, M = o.useCallback(
      async (n) => {
        if (se(n.validity)) {
          const i = B(n.validity);
          C(u, i);
          return;
        }
        const F = n.form ? new FormData(n.form) : new FormData(), b = [n.value, F], R = [], A = [];
        x.forEach((i) => {
          Me(i, b) ? A.push(i) : be(i) && R.push(i);
        });
        const w = R.map(({ id: i, match: E }) => [i, E(...b)]), T = Object.fromEntries(w), k = Object.values(T).some(Boolean), r = k;
        n.setCustomValidity(r ? V : "");
        const m = B(n.validity);
        if (C(u, m), y(u, T), !k && A.length > 0) {
          const i = A.map(
            ({ id: ie, match: ce }) => ce(...b).then((le) => [ie, le])
          ), E = await Promise.all(i), Q = Object.fromEntries(E), ne = Object.values(Q).some(Boolean);
          n.setCustomValidity(ne ? V : "");
          const ae = B(n.validity);
          C(u, ae), y(u, Q);
        }
      },
      [x, u, y, C]
    );
    o.useEffect(() => {
      const n = d.current;
      if (n) {
        const F = () => M(n);
        return n.addEventListener("change", F), () => n.removeEventListener("change", F);
      }
    }, [M]);
    const g = o.useCallback(() => {
      const n = d.current;
      n && (n.setCustomValidity(""), I(u));
    }, [u, I]);
    o.useEffect(() => {
      const n = d.current?.form;
      if (n)
        return n.addEventListener("reset", g), () => n.removeEventListener("reset", g);
    }, [g]), o.useEffect(() => {
      const n = d.current, F = n?.closest("form");
      if (F && l.serverInvalid) {
        const b = re(F);
        b === n && b.focus();
      }
    }, [l.serverInvalid]);
    const _ = a.getFieldValidity(u);
    return /* @__PURE__ */ h.jsx(
      j.input,
      {
        "data-valid": z(_, l.serverInvalid),
        "data-invalid": J(_, l.serverInvalid),
        "aria-invalid": l.serverInvalid ? !0 : void 0,
        "aria-describedby": f.getFieldDescription(u),
        title: "",
        ...c,
        ref: v,
        id: p,
        name: u,
        onInvalid: S(e.onInvalid, (n) => {
          const F = n.currentTarget;
          M(F);
        }),
        onChange: S(e.onChange, (n) => {
          g();
        })
      }
    );
  }
);
ee.displayName = D;
var V = "This value is not valid", he = {
  badInput: V,
  patternMismatch: "This value does not match the required pattern",
  rangeOverflow: "This value is too large",
  rangeUnderflow: "This value is too small",
  stepMismatch: "This value does not match the required step",
  tooLong: "This value is too long",
  tooShort: "This value is too short",
  typeMismatch: "This value does not match the required type",
  valid: void 0,
  valueMissing: "This value is missing"
}, P = "FormMessage", te = o.forwardRef(
  (e, t) => {
    const { match: s, name: c, ...a } = e, l = K(P, e.__scopeForm), f = c ?? l.name;
    return s === void 0 ? /* @__PURE__ */ h.jsx($, { ...a, ref: t, name: f, children: e.children || V }) : typeof s == "function" ? /* @__PURE__ */ h.jsx(Fe, { match: s, ...a, ref: t, name: f }) : /* @__PURE__ */ h.jsx(Ce, { match: s, ...a, ref: t, name: f });
  }
);
te.displayName = P;
var Ce = o.forwardRef(
  (e, t) => {
    const { match: s, forceMatch: c = !1, name: a, children: l, ...f } = e, v = L(P, f.__scopeForm).getFieldValidity(a);
    return c || v?.[s] ? /* @__PURE__ */ h.jsx($, { ref: t, ...f, name: a, children: l ?? he[s] }) : null;
  }
), Fe = o.forwardRef(
  (e, t) => {
    const { match: s, forceMatch: c = !1, name: a, id: l, children: f, ...d } = e, v = L(P, d.__scopeForm), u = o.useRef(null), p = N(t, u), x = U(), C = l ?? x, y = o.useMemo(() => ({ id: C, match: s }), [C, s]), { onFieldCustomMatcherEntryAdd: I, onFieldCustomMatcherEntryRemove: M } = v;
    o.useEffect(() => (I(a, y), () => M(a, y.id)), [y, a, I, M]);
    const g = v.getFieldValidity(a), n = v.getFieldCustomErrors(a)[C];
    return c || g && !se(g) && n ? /* @__PURE__ */ h.jsx($, { id: C, ref: p, ...d, name: a, children: f ?? V }) : null;
  }
), $ = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, id: c, name: a, ...l } = e, f = W(P, s), d = U(), v = c ?? d, { onFieldMessageIdAdd: u, onFieldMessageIdRemove: p } = f;
    return o.useEffect(() => (u(a, v), () => p(a, v)), [a, v, u, p]), /* @__PURE__ */ h.jsx(j.span, { id: v, ...l, ref: t });
  }
), ye = "FormSubmit", oe = o.forwardRef(
  (e, t) => {
    const { __scopeForm: s, ...c } = e;
    return /* @__PURE__ */ h.jsx(j.button, { type: "submit", ...c, ref: t });
  }
);
oe.displayName = ye;
function B(e) {
  const t = {};
  for (const s in e)
    t[s] = e[s];
  return t;
}
function Ee(e) {
  return e instanceof HTMLElement;
}
function ge(e) {
  return "validity" in e;
}
function pe(e) {
  return ge(e) && (e.validity.valid === !1 || e.getAttribute("aria-invalid") === "true");
}
function re(e) {
  const t = e.elements, [s] = Array.from(t).filter(Ee).filter(pe);
  return s;
}
function Me(e, t) {
  return e.match.constructor.name === "AsyncFunction" || Ie(e.match, t);
}
function be(e) {
  return e.match.constructor.name === "Function";
}
function Ie(e, t) {
  return e(...t) instanceof Promise;
}
function se(e) {
  let t = !1;
  for (const s in e) {
    const c = s;
    if (c !== "valid" && c !== "customError" && e[c]) {
      t = !0;
      break;
    }
  }
  return t;
}
function z(e, t) {
  if (e?.valid === !0 && !t) return !0;
}
function J(e, t) {
  if (e?.valid === !1 || t) return !0;
}
var Te = X, ke = Y, De = Z, we = ee, Be = te, Oe = oe;
export {
  we as Control,
  ke as Field,
  X as Form,
  ee as FormControl,
  Y as FormField,
  Z as FormLabel,
  te as FormMessage,
  oe as FormSubmit,
  De as Label,
  Be as Message,
  Te as Root,
  Oe as Submit,
  Pe as createFormScope
};
