// @estarlincito/utils@2.4.2/dist/form-zed.js
import { handleError as a } from "./handle-error.js";
const i = {
  "first-name": {
    placeholder: "Estarlin",
    type: "text",
    maxLength: 30,
    minLength: 1,
    title: ""
  },
  "last-name": {
    placeholder: "R",
    type: "text",
    maxLength: 30,
    minLength: 1,
    title: ""
  },
  company: {
    placeholder: "Estarlincito Support's",
    type: "text",
    maxLength: 30,
    minLength: 2,
    title: ""
  },
  "phone-number": {
    placeholder: "123-456-7890",
    type: "tel",
    maxLength: 12,
    minLength: 10,
    title: ""
  },
  email: {
    placeholder: "name@domain.com",
    type: "email",
    maxLength: 30,
    minLength: 5,
    title: ""
  },
  password: {
    placeholder: "4ppl3-1Ph0ne",
    type: "password",
    maxLength: 30,
    minLength: 8,
    title: ""
  },
  message: {
    placeholder: "Enter your message here…",
    type: "text",
    maxLength: 1e3,
    minLength: 1,
    title: ""
  },
  username: {
    placeholder: "estarlincito",
    type: "text",
    maxLength: 20,
    minLength: 5,
    title: ""
  }
}, o = (e) => {
  const t = e[0], l = e.replace(t, t.toLocaleUpperCase()).replace("-", " "), n = i[e];
  if (!n) {
    a(`${e} was not found in (fields) /lib/zed.ts`);
    return;
  }
  return { ...n, title: l };
};
export {
  o as formZed
};
